deployment/hasura/migrations/AerieMerlin/2_drop_profile_topic_fks/up.sql

```diff
@@ -0,0 +1,286 @@
+-- profile_segment
+alter table profile_segment drop constraint profile_segment_owned_by_profile;
+
+create or replace function profile_segment_integrity_function()
+  returns trigger
+  security invoker
+  language plpgsql as $$begin
+  if not exists(
+      select from profile
+      where profile.dataset_id = new.dataset_id
+        and profile.id = new.profile_id
+        for key share of profile)
+    -- for key share is important: it makes sure that concurrent transactions cannot update
+    -- the columns that compose the profile's key until after this transaction commits.
+  then
+    raise exception 'foreign key violation: there is no profile with id % in dataset %', new.profile_id, new.dataset_id;
+  end if;
+  return new;
+end$$;
+
+comment on function profile_segment_integrity_function is e''
+  'Used to simulate a foreign key constraint between profile_segment and profile, to avoid acquiring a lock on the'
+  'profile table when creating a new partition of profile_segment. This function checks that a corresponding profile'
+  'exists for every inserted or updated profile_segment. A trigger that calls this function is added separately to each'
+  'new partition of profile_segment.';
+
+create constraint trigger insert_update_profile_segment_trigger
+  after insert or update on profile_segment
+  for each row
+execute function profile_segment_integrity_function();
+
+-- event
+alter table event drop constraint event_owned_by_topic;
+
+create or replace function event_integrity_function()
+  returns trigger
+  security invoker
+  language plpgsql as $$begin
+  if not exists(
+      select from topic
+      where topic.dataset_id = new.dataset_id
+        and topic.topic_index = new.topic_index
+        for key share of topic)
+    -- for key share is important: it makes sure that concurrent transactions cannot update
+    -- the columns that compose the topic's key until after this transaction commits.
+  then
+    raise exception 'foreign key violation: there is no topic with topic_index % in dataset %', new.topic_index, new.dataset_id;
+  end if;
+  return new;
+end$$;
+
+comment on function event_integrity_function is e''
+  'Used to simulate a foreign key constraint between event and topic, to avoid acquiring a lock on the'
+  'topic table when creating a new partition of event. This function checks that a corresponding topic'
+  'exists for every inserted or updated event. A trigger that calls this function is added separately to each'
+  'new partition of event.';
+
+create constraint trigger insert_update_event_trigger
+  after insert or update on event
+  for each row
+execute function event_integrity_function();
+
+-- profile
+create or replace function delete_profile_cascade()
+  returns trigger
+  security invoker
+  language plpgsql as $$begin
+  delete from profile_segment
+  where profile_segment.dataset_id = old.dataset_id and profile_segment.profile_id = old.id;
```

Comment:
Indentation differs between here and `profile.sql`

Reply:
Thanks - I ran a diff and I think I've ironed out the indentation differences. (There are some differences left over from anchors - but I won't touch those in this PR)

Reply:
I've double-checked all of my "indentation differs" comments and left the three that still apply.

Reply:
Ah, I see now, those indentation differences are not in the code of a function, so they won't cause problems in a diff :+1: (The reason I know that is I still have my local code to do the pg_dump diff) I'll adjust that indentation for the sake of consistency, but I don't believe it will cause any issues.

Reply:
Ah, no, no need! I complete forgot that it only matters in the body of the function!